### PR TITLE
DB-6373: set rowlocation for index lookup

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

File: splice_machine/src/test/java/com/splicemachine/derby/impl/load/HBaseBulkLoadIT.java [hunk 1]
```diff
@@ -101,22 +101,23 @@ public static void cleanup() throws Exception {
...
         countUsingIndex(LINEITEM, "L_SHIPDATE_IDX");
         countUsingIndex(ORDERS, "O_CUST_IDX");
         countUsingIndex(ORDERS, "O_DATE_PRI_KEY_IDX");
     }
 
-    private static void bulkDelete(String tableName) throws Exception {
-        String sql = String.format("delete from %s --splice-properties bulkDeleteDirectory='%s'", tableName, getResource("data"));
+    private static void bulkDelete(String tableName, String indexName) throws Exception {
+        String sql = String.format("delete from %s --splice-properties bulkDeleteDirectory='%s',index=%s", tableName, getResource("data"),
+                indexName!=null?indexName:"null");
```
Comment (snowspeeder) [inline splice_machine/src/test/java/com/splicemachine/derby/impl/load/HBaseBulkLoadIT.java]: Could you also add IT for regular delete with index hint?

Comment (jyuanca) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/load/HBaseBulkLoadIT.java]: Regular delete was not impacted.

**Comment by snowspeeder** [comment]
regular delete with index hint will not hit this issue?

File: splice_machine/src/test/java/com/splicemachine/derby/impl/load/HBaseBulkLoadIT.java (hunk 1 above)
Comment (jyuanca) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/load/HBaseBulkLoadIT.java]: It only happens when using results from index lookup to create index using IndexTransformer. Bulk delete is the only case

**Comment by snowspeeder** [comment]
I see, thanks for the clarification.

**Pull request merged**